src/libraries/PositionConfigId.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.24;
+
+import {PositionConfig, PositionConfigLibrary} from "./PositionConfig.sol";
+
+/// @notice A configId is set per tokenId
+/// The lower 255 bits are used to store the truncated hash of the corresponding PositionConfig
+/// The upper bit is used to signal if the tokenId has a subscriber
+struct PositionConfigId {
+    bytes32 id;
+}
+
+library PositionConfigIdLibrary {
+    using PositionConfigLibrary for PositionConfig;
+
+    bytes32 constant MASK_UPPER_BIT = 0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
+    bytes32 constant DIRTY_UPPER_BIT = 0x8000000000000000000000000000000000000000000000000000000000000000;
+
+    /// @notice returns the truncated hash of the PositionConfig for a given tokenId
+    function getConfigId(PositionConfigId storage _configId) internal view returns (bytes32 configId) {
+        configId = _configId.id & MASK_UPPER_BIT;
+    }
+
+    /// @dev We only set the config on mint, guaranteeing that the most significant bit is unset, so we can just assign the entire 32 bytes to the id.
+    function setConfigId(PositionConfigId storage configId, PositionConfig calldata config) internal {
+        configId.id = config.toId();
```

Comment:
only question for me is whether this function should be

```
setConfigId(PositionConfigId storage configId, bytes32 id)
configId.id = id;
```

and then mint would do
```
positionConfigs[tokenId].setConfigId(config.toId());
```

thoughts? its not a strong opinion just a nit thought

Reply:
and then it would remove the imports of `PositionConfig` and `PositionConfigLibrary` in this file

Reply:
yeahh I thought about that to.. I can change it